src/preprocessing/particle_packing/system.jl

```diff
@@ -246,7 +254,7 @@ function kinetic_energy(system::ParticlePackingSystem, v_ode, u_ode, semi, t)
 
     # If `each_moving_particle` is empty (no moving particles), return zero
     return sum(each_moving_particle(system), init=zero(eltype(system))) do particle
-        velocity = current_velocity(system.initial_condition.velocity, system, particle)
+        velocity = current_velocity(system.advection_velocity, system, particle)
```

Comment:
It's confusing to misuse `current_velocity` like this. Please create a separate wrapper function
```julia
@inline advection_velocity(system::PackingSystem, particle) = extract_svector(system.advection_velocity, system, particle)
```